streams/src/test/java/org/apache/kafka/streams/integration/GlobalKTableIntegrationTest.java

```diff
@@ -217,6 +226,37 @@ public boolean conditionMet() {
         }, 30000L, "waiting for final values");
     }
 
+    @Test
+    public void shouldRestoreTransactionalMessages() throws Exception {
+        produceInitialGlobalTableValues(true);
+        startStreams();
```

Comment:
@alexjg The code is actually looping in this line, before the `waitForCondition` call actually, that is why you are seeing the indefinite hanging.